docs/developer/pull-request.md

```diff
@@ -0,0 +1,107 @@
+# Pull Requests
+
+## Submitter Guidelines
+
+- Fill in [our pull request template](../../.github/PULL_REQUEST_TEMPLATE.md). In addition, write a clear and concise PR
```

Comment:
how is it different from the details that will be in the corresponding issue?

Reply:
It should be a summary of the changes. It may repeat some of the text in the story but it's still nice to have a description in the PR so you don't have to constantly refer back to the issue.

Reply:
shall we acknowledge that?

For example:

Fill in [our pull request template](https://github.com/nginxinc/nginx-kubernetes-gateway/blob/12bb87c3ad247a60b32a06bf9bb6c2b0b8b5bfd6/.github/PULL_REQUEST_TEMPLATE.md). In addition, write a clear and concise PR description that helps reviewers understand the purpose and impact of your changes. Start with a brief overview of the problem or feature being addressed _and reference the issue that the PR implements._ Then, explain the approach you took to implement the solution, highlighting any significant design decisions or considerations. Finally, mention any specific areas where you would like reviewers to focus their attention or provide specific feedback. A well-written PR description sets the stage for effective code review and facilitates a smoother review process.

also, do we also need to change the template, so that it asks for those things?

Reply:
The next line says to include the issue number in the PR description so I don't think we need add that to this paragraph.

The template already has this:
> Describe the use case and detail of the change. If this PR addresses an issue on GitHub, make sure to include a link to that issue here in this description (not in the title of the PR).

Reply:
Decided to move the description to the pull request template and just link the template here. I also added the problem/solution/testing format. Let me know what you think